artifactory.py

```diff
@@ -997,11 +997,30 @@ def deploy(
         parameters=None,
         explode_archive=None,
         explode_archive_atomic=None,
+        checksum=None,
+        by_checksum=None,
     ):
         """
         Uploads a given file-like object
         HTTP chunked encoding will be attempted
+
+        If by_checksum is True, fobj should be None
+
+        :param pathobj: ArtifactoryPath object
+        :param fobj: file object to be deployed
+        :param md5:
+        :param sha1:
+        :param sha256:
+        :param parameters: Artifact properties
+        :param explode_archive(bool): True: archive will be exploded upon deployment
+        :param explode_archive_atomic(bool): True: archive will be exploded in an atomic operation upon deployment
+        :param checksum: sha1Value or sha256Value
```

Comment:
```suggestion
        :param checksum_val: sha1Value or sha256Value
```

Reply:
The naming of `checksum` is following the same rule of `sha1` and `sha256`.

Reply:
@zhan9san 
this is a new parameter that was not present in the library. Do not forget about python Zen: Explicit is better than implicit.

Reply:
Hi @beliaev-maksim 
Another one is `Simple is better than complex`.

Would it be possible to show some ambiguities of `checksum`?

Reply:
adding `_val` does not increase complexity, this is not a valid point

if you add `_val`, you super explicitly show that you expect checksum value and not something else, user will not need to go to help and check what you mean by this

Reply:
It's not as simple as `checksum`.

> not something else,

Could you please list what something else it can be?

I can't believe there are someone who use this Python lib without checking [DeployArtifactbyChecksum](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-DeployArtifactbyChecksum)

And the official docs shows us `Deploy Artifact by Checksum`.
With respect, do you think `Checksum -> checksum_val` is more explicit than `Checksum -> checksum`?

Reply:
we have a mix of functions that request calculation of checksum vs checksum themselves.
but point to be consistent with REST API name makes sense